lib/std/mem.zig

```diff
@@ -116,9 +116,6 @@ pub const Allocator = struct {
         if (isAligned(@ptrToInt(old_mem.ptr), new_alignment)) {
             if (new_byte_count <= old_mem.len) {
                 const shrunk_len = self.shrinkBytes(old_mem, new_byte_count, len_align);
-                if (shrunk_len < old_mem.len) {
```

Comment:
I mistakenly added this when I implemented the new allocator interface.  We cannot `@memset` after shrinking memory, it must be done beforehand and might be best to do this in the allocators themselves.

Reply:
Related: #4298